addons/crm/models/crm_lead.py

```diff
@@ -595,7 +595,16 @@ def write(self, vals):
         if any(field in ['active', 'stage_id'] for field in vals):
             self._handle_won_lost(vals)
 
-        write_result = super(Lead, self).write(vals)
+        # stage change between two won stages: does not chang the date_closed
+        if stage_is_won:
+            still_won_leads = self.filtered(lambda r: r.stage_id and r.stage_id.is_won)
+            new_won_leads = self - still_won_leads
+            still_won_vals = vals.copy()
+            del still_won_vals['date_closed']
+            write_result = super(Lead, still_won_leads).write(still_won_vals)
+            write_result += super(Lead, new_won_leads).write(vals)
```

Comment:
Writer always returns True so no need to "plus"' it, we just have to return True, or the result to any call to super.